File: frontend/data/models/order.js
class Order{
    constructor(orderID, customerID, storeID, orderDate, nItems, total) {
        this.orderID = orderID;
        this.customerID = customerID;
        this.storeID = storeID;
        this.orderDate = orderDate;
        this.nItems = nItems;
        this.total = total;
    }

    
    static fromJSON(json) {
        
        if (!json || typeof json !== 'object' ||
            !json.hasOwnProperty('orderID') ||
            !json.hasOwnProperty('customerID') ||
            !json.hasOwnProperty('storeID') ||
            !json.hasOwnProperty('orderDate') ||
            !json.hasOwnProperty('nItems') ||
            !json.hasOwnProperty('total')) {
            throw new Error('Invalid JSON data for Order');
        }

        return new Order(
            json.orderID,
            json.customerID,
            json.storeID,
            json.orderDate,
            json.nItems,
            json.total
        );
    }
}

File: backend/api/Models/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;

namespace api.Models
{
    public class Order
    {
       public int OrderId { get; set; }
       public string CustomerId { get; set; } = string.Empty;
       public string StoreId { get; set; } = string.Empty;
       public DateTime OrderDate { get; set; }
       public int NItems { get; set; }
       public double total { get; set; }

       //public string StoreId { get; set;} = string.Empty;
       public Store? Store { get; set;}

       public Customer? customer { get; set; }

      public List<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    }
}
